dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/DebugHelper.java

```diff
@@ -0,0 +1,337 @@
+            + nameValue("elapsed-time", formatNanoseconds(finishState.wallClock() - startState.wallClock()));
+    }
+
+    public static String formatNanoseconds(long ns) {
+        return String.format("%.3f sec", ns / 1e9);
+    }
+
+    public static String formatBytes(long bytes) {
+        return String.format("%.2f MB", bytes / 1e6);
+    }
+
+    private static String formatMemoryPoolMXBean(MemoryPoolMXBean pool) {
+        return nameValue("POOL", pool.getName() + ", " + pool.getType())
+            + nameValue("current", pool.getUsage())
+            + nameValue("peak", pool.getPeakUsage())
+            + nameValue("collection", pool.getCollectionUsage());
+    }
+
+    private static String nameValue(String name, Object value) {
+        return String.format("%s: %s\n", name, value);
+    }
+
+    private static <T> List<String> format(Function<T, String> formatter, Collection<T> objects) {
+        return objects.stream().map(formatter).toList();
+    }
+
+    private static String concat(Collection<?> objects) {
+        return objects.stream().map(Object::toString).collect(Collectors.joining());
+    }
+
+    private static ThreadState getState() {
+        long bytes = 0;
+        if (threadMXBean instanceof com.sun.management.ThreadMXBean sunBean) {
+            bytes = sunBean.getCurrentThreadAllocatedBytes();
+        } else {
+            LOG.info("sun threadMXBean.getCurrentThreadAllocatedBytes() not supported");
+        }
+        long cpuTime = 0;
+        try {
+            cpuTime = threadMXBean.getCurrentThreadCpuTime();
+        } catch (UnsupportedOperationException e) {
+            LOG.info("threadMXBean.getCurrentThreadCpuTime not supported");
+        }
+        long userTime = 0;
+        try {
+            userTime = threadMXBean.getCurrentThreadUserTime();
+        } catch (UnsupportedOperationException e) {
+            LOG.info("threadMXBean.getCurrentThreadUserTime not supported");
+        }
+        long wallClock = System.nanoTime();
+        return new ThreadState(bytes, cpuTime, userTime, wallClock);
+    }
+
+    private static Optional<Session> getCurrentSession() {
+        try {
+            // This getCurrentSession() call will throw if there's no current session.
+            return Optional.of(sessionFactory.getCurrentSession());
+        } catch (Exception e) {
+            return Optional.empty();
+        }
+    }
+
+    private static String formatRequest(ContainerRequest request) {
+        return String.format("%s \"%s\"", request.getMethod(), request.getPath(false));
+    }
+
+    private static void handleRequestEvent(RequestEvent event, ThreadState startState) {
+        try {
+            ContainerRequest request = event.getContainerRequest();
+            ContainerResponse response = event.getContainerResponse();
+
+            switch (event.getType()) {
+
+            // Request started.
+            case START:
+                logStarted(request);
```

Comment:
This will be very useful; right now the request is only logged when completed. If we die (rare, but it's happened) processing a request, it's hard to piece together what the request was.